src/boot/efi/stub.c

```diff
@@ -76,10 +75,17 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
                 cmdline = line;
 
 #if ENABLE_TPM
+        } else if (cmdline_len > 0) {
+                UINTN i;
+                options = AllocatePool(cmdline_len * sizeof(CHAR16));
+                for (i= 0; i < cmdline_len; i++)
+                        options[i] = cmdline[0];
+        }
+        if (options) {
                 /* Try to log any options to the TPM, especially manually edited options */
                 err = tpm_log_event(SD_TPM_PCR,
-                                    (EFI_PHYSICAL_ADDRESS) (UINTN) loaded_image->LoadOptions,
-                                    loaded_image->LoadOptionsSize, loaded_image->LoadOptions);
+                                    (EFI_PHYSICAL_ADDRESS) (UINTN) options,
+                                    cmdline_len, options);
```

Comment:
passing cmdline_len is wrong here, in case the first if branch is taken, it will be the correct value halved...